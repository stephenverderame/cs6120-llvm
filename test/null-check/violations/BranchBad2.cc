
#include <cstdio>
#include <cstdlib>
int main()
{
    // NOLINTNEXTLINE
    auto P = (double*)malloc(sizeof(double));
    if (P != nullptr) {
        *P = 1.0;
    }
    *P = 2.0;
}