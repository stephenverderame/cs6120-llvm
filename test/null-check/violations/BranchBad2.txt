[31mSafety Violation[00m: Use of potentially null pointer in store at BranchBad2.cc:11
clang++-17: error: clang frontend command failed with exit code 2 (use -v to see invocation)
