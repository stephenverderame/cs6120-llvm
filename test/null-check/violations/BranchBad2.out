[31mSafety Violation[00m: Use of potentially null pointer in store at BranchBad2.cc:11
